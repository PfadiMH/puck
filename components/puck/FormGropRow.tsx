import { Input } from "@components/misc/external/input";
import { FormField } from "./FormGroup";

export function FormGroupRow({ row }: { row: FormField[] }) {
  return (
    <div className='flex'>
      {row.map((field, index) => (
        <span className='flex grow mt-2 mb-2' key={`field-${index}`}>
          <span className='flex grow'>
            <span
              className={`${field.size === "full" ? "flex-1" : "flex-1/2"} ${
                index === 0 ? "mr-2" : ""
              }`}
            >
              <label htmlFor={field.label}>{field.label}</label>
              <Input
                id={field.label}
                name={field.label}
                placeholder={field.placeholder}
                required={field.required}
                type={field.type}
              />
            </span>
          </span>
        </span>
      ))}
    </div>
  );
}
