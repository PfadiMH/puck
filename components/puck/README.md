# /components/puck

This directory contains React components specifically designed for use as building blocks within the Puck editor.

**Key points:**

- **Important:** All components must be registered in `lib/config/page.config.ts` to be available in the Puck editor.
- [Puck Docs on Defining Components](https://puckeditor.com/docs/integrating-puck/component-configuration)

### Component Configuration

Each component within this directory must export two key items for it to be registered and utilized in `page.config.ts`: a `config` object and a `config` type.

This `config` specifies how the component should be rendered and what fields are available for editing.

- `render`: This property is a function that returns the React component to be rendered. It receives the props as defined in the `fields` object.
- `fields`: This is an object that defines the editable fields for the component in the Puck editor. The value is an object that specifies the type of input to be used in the editor (e.g., "text", "select").
