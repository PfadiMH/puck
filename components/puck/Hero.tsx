import { PostHeroSvg } from "@components/graphics/PostHeroSvg";
import {
  FileProps,
  singleSelectionFileTableField,
} from "@components/puck-fields/fileTable";
import { ComponentConfig } from "@measured/puck";
import Image from "next/image";

export type HeroProps = {
  title: string;
  backgroundImage: FileProps;
};

function Hero({ title, backgroundImage }: HeroProps) {
  const imageUrl = backgroundImage?.url;

  return (
    <div className="full w-full h-96 relative flex flex-col justify-center overflow-hidden items-center">
      {imageUrl && (
        <Image
          fill
          src={imageUrl}
          alt="Hero Image"
          style={{ objectFit: "cover" }}
        />
      )}
      {title && (
        <>
          <h1 className="text-4xl font-bold text-center z-10 text-white">
            {title}
          </h1>
          <div className="bg-black opacity-15 absolute w-full h-full" />
        </>
      )}
      <PostHeroSvg className="absolute bottom-0" />
    </div>
  );
}

export const heroConfig: ComponentConfig<HeroProps> = {
  render: Hero,
  fields: {
    title: {
      type: "text",
      label: "Title (Optional)",
    },
    backgroundImage: singleSelectionFileTableField,
  },
};
