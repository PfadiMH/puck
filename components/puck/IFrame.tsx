import { ComponentConfig } from "@measured/puck";
import React from "react";

export type IFrameProps = {
  source: string;
  height: string;
  title: string;
};

function IFrame({ source, height, title }: IFrameProps) {
  return (
    <>
      <div className="w-full overflow-hidden">
        <iframe src={source} width="100%" height={height} title={title} />
      </div>
    </>
  );
}

export const iframeConfig: ComponentConfig<IFrameProps> = {
  render: IFrame,
  fields: {
    source: {
      type: "textarea",
    },
    height: {
      type: "number",
    },
    title: {
      type: "textarea",
    },
  },
};
