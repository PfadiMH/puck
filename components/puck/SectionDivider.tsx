import { SectionBreak } from "@components/misc/SectionBreak";
import { ComponentConfig } from "@measured/puck";

export type SectionDividerProps = {};

function SectionDivider({}: SectionDividerProps) {
  return <SectionBreak />;
}

export const sectionDividerConfig: ComponentConfig<SectionDividerProps> = {
  render: SectionDivider,
};
