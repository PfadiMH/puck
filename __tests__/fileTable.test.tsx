import {
  multipleSelectionFileTableField,
  singleSelectionFileTableField,
} from "@components/puck-fields/fileTable";
import { PuckContext } from "@measured/puck";
import "@testing-library/jest-dom";
import { render, screen, within } from "@testing-library/react";

import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

jest.mock("@lib/filemanager/filemanager", () => {
  const mockFileManagerService = {
    deleteFile: jest.fn(),
    saveFile: jest.fn(),
    getFileUrl: jest.fn((name) => {
      return Promise.resolve(`https://mock-url.com/${name}`);
    }),
    getFile: jest.fn(),
    getFileNames: jest.fn(() => {
      return Promise.resolve([
        "test-file.txt",
        "another-file.png",
        "document.pdf",
      ]);
    }),
  };

  return {
    getFileManagerService: jest.fn(() =>
      Promise.resolve(mockFileManagerService)
    ),
  };
});

describe("singleSelectionFileTable component", () => {
  const renderWithQueryClient = (ui: React.ReactElement) => {
    return render(
      <QueryClientProvider client={queryClient}>{ui}</QueryClientProvider>
    );
  };

  beforeEach(() => {
    queryClient.clear();
  });

  it("renders the component unchanged (snapshot)", async () => {
    const { container } = renderWithQueryClient(
      <singleSelectionFileTableField.render
        {...{
          field: singleSelectionFileTableField,
          id: "test-id",
          name: "test-name",
          value: {
            name: "test-file.txt",
            url: "https://mock-url.com/test-file.txt",
          },
          onChange: () => {},
          puck: {} as PuckContext,
        }}
      />
    );
    await screen.findByText("test-file.txt");
    expect(container).toMatchSnapshot();
  });

  it("renders a singleSelectionFileTable", async () => {
    renderWithQueryClient(
      <singleSelectionFileTableField.render
        {...{
          field: singleSelectionFileTableField,
          id: "test-id",
          name: "test-name",
          value: {
            name: "test-file.txt",
            url: "https://mock-url.com/test-file.txt",
          },
          onChange: () => {},
          puck: {} as PuckContext,
        }}
      />
    );
    expect(screen.getByText("loading...")).toBeInTheDocument();
    const textElement = await screen.findByText("test-file.txt");
    expect(textElement).toBeInTheDocument();
    expect(screen.queryByText("loading...")).not.toBeInTheDocument();
  });

  it("marks the file specified in 'value' as selected", async () => {
    const fileNameToTest = "test-file.txt";
    const fileUrlToTest = "https://test-url.com";

    renderWithQueryClient(
      <singleSelectionFileTableField.render
        {...{
          field: singleSelectionFileTableField,
          id: "test-id",
          name: "test-name",
          value: { name: fileNameToTest, url: fileUrlToTest },
          onChange: () => {},
          puck: {} as PuckContext,
        }}
      />
    );

    const fileNameElement = await screen.findByText(fileNameToTest);
    expect(fileNameElement).toBeInTheDocument();

    const rowElement = fileNameElement.closest("tr");
    expect(rowElement).toBeInTheDocument();

    expect(rowElement).toHaveAttribute("data-state", "selected");

    const radioButton = within(rowElement!).getByRole("radio");
    expect(radioButton).toBeInTheDocument();
    expect(radioButton).toBeChecked();
    expect(radioButton).toHaveAttribute("aria-checked", "true");
  });
});

describe("multipleSelectionFileTable component", () => {
  const renderWithQueryClient = (ui: React.ReactElement) => {
    return render(
      <QueryClientProvider client={queryClient}>{ui}</QueryClientProvider>
    );
  };

  beforeEach(() => {
    queryClient.clear();
  });
  it("renders the component unchanged (snapshot)", async () => {
    const { container } = renderWithQueryClient(
      <multipleSelectionFileTableField.render
        {...{
          field: multipleSelectionFileTableField,
          id: "test-id",
          name: "test-name",
          value: [
            {
              name: "test-file.txt",
              url: "https://mock-url.com/test-file.txt",
            },
            {
              name: "another-file.png",
              url: "https://mock-url.com/another-file.png",
            },
          ],
          onChange: () => {},
          puck: {} as PuckContext,
        }}
      />
    );
    await screen.findByText("test-file.txt");
    expect(container).toMatchSnapshot();
  });

  it("renders a multipleSelectionFileTableField", async () => {
    renderWithQueryClient(
      <multipleSelectionFileTableField.render
        {...{
          field: multipleSelectionFileTableField,
          id: "test-id",
          name: "test-name",
          value: [
            {
              name: "test-file.txt",
              url: "https://mock-url.com/test-file.txt",
            },
            {
              name: "another-file.png",
              url: "https://mock-url.com/another-file.png",
            },
          ],
          onChange: () => {},
          puck: {} as PuckContext,
        }}
      />
    );
    expect(screen.getByText("loading...")).toBeInTheDocument();
    const textElement = await screen.findByText("test-file.txt");
    expect(textElement).toBeInTheDocument();
    expect(screen.queryByText("loading...")).not.toBeInTheDocument();
  });
});
