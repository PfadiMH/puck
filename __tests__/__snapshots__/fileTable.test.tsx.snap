// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`multipleSelectionFileTable component renders the component unchanged (snapshot) 1`] = `
<div>
  <div
    class="flex flex-col w-full h-full gap-4"
  >
    <input
      class="flex h-9 w-full rounded-md border border-cn-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
      multiple=""
      type="file"
    />
    <div
      aria-required="false"
      class="grid gap-2"
      dir="ltr"
      role="radiogroup"
      style="outline: none;"
      tabindex="-1"
    >
      <div
        class="flex flex-col gap-5 w-min"
      >
        <input
          class="flex h-9 w-full rounded-md border border-cn-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm max-w-sm"
          placeholder="Search ..."
          value=""
        />
        <div
          class="rounded-md border"
        >
          <div
            class="relative w-full overflow-auto"
          >
            <table
              class="w-full caption-bottom text-sm"
            >
              <thead
                class="[&_tr]:border-b"
              >
                <tr
                  class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <th
                    class="h-10 px-2 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <button
                      aria-checked="mixed"
                      aria-label="Select all"
                      class="peer h-4 w-4 shrink-0 rounded-sm border border-cn-primary shadow focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-cn-ring disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-cn-primary data-[state=checked]:text-cn-primary-foreground"
                      data-state="indeterminate"
                      role="checkbox"
                      type="button"
                      value="on"
                    >
                      <span
                        class="flex items-center justify-center text-current"
                        data-state="indeterminate"
                        style="pointer-events: none;"
                      >
                        <svg
                          class="lucide lucide-check h-4 w-4"
                          fill="none"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M20 6 9 17l-5-5"
                          />
                        </svg>
                      </span>
                    </button>
                  </th>
                  <th
                    class="h-10 px-2 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <div
                      class="text-right"
                    >
                      Filename
                    </div>
                  </th>
                  <th
                    class="h-10 px-2 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  />
                </tr>
              </thead>
              <tbody
                class="[&_tr:last-child]:border-0"
              >
                <tr
                  class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                  data-state="selected"
                >
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <button
                      aria-checked="true"
                      aria-label="Select row"
                      class="peer h-4 w-4 shrink-0 rounded-sm border border-cn-primary shadow focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-cn-ring disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-cn-primary data-[state=checked]:text-cn-primary-foreground"
                      data-state="checked"
                      role="checkbox"
                      type="button"
                      value="on"
                    >
                      <span
                        class="flex items-center justify-center text-current"
                        data-state="checked"
                        style="pointer-events: none;"
                      >
                        <svg
                          class="lucide lucide-check h-4 w-4"
                          fill="none"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M20 6 9 17l-5-5"
                          />
                        </svg>
                      </span>
                    </button>
                  </td>
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <div
                      class="text-right font-medium whitespace-normal break-words"
                    >
                      test-file.txt
                    </div>
                  </td>
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <button
                      class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-cn-primary underline-offset-4 hover:underline h-8 rounded-md px-3 text-xs"
                    >
                      <svg
                        class="lucide lucide-trash2 lucide-trash-2"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3 6h18"
                        />
                        <path
                          d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                        />
                        <path
                          d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                        />
                        <line
                          x1="10"
                          x2="10"
                          y1="11"
                          y2="17"
                        />
                        <line
                          x1="14"
                          x2="14"
                          y1="11"
                          y2="17"
                        />
                      </svg>
                    </button>
                  </td>
                </tr>
                <tr
                  class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                  data-state="selected"
                >
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <button
                      aria-checked="true"
                      aria-label="Select row"
                      class="peer h-4 w-4 shrink-0 rounded-sm border border-cn-primary shadow focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-cn-ring disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-cn-primary data-[state=checked]:text-cn-primary-foreground"
                      data-state="checked"
                      role="checkbox"
                      type="button"
                      value="on"
                    >
                      <span
                        class="flex items-center justify-center text-current"
                        data-state="checked"
                        style="pointer-events: none;"
                      >
                        <svg
                          class="lucide lucide-check h-4 w-4"
                          fill="none"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M20 6 9 17l-5-5"
                          />
                        </svg>
                      </span>
                    </button>
                  </td>
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <div
                      class="text-right font-medium whitespace-normal break-words"
                    >
                      another-file.png
                    </div>
                  </td>
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <button
                      class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-cn-primary underline-offset-4 hover:underline h-8 rounded-md px-3 text-xs"
                    >
                      <svg
                        class="lucide lucide-trash2 lucide-trash-2"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3 6h18"
                        />
                        <path
                          d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                        />
                        <path
                          d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                        />
                        <line
                          x1="10"
                          x2="10"
                          y1="11"
                          y2="17"
                        />
                        <line
                          x1="14"
                          x2="14"
                          y1="11"
                          y2="17"
                        />
                      </svg>
                    </button>
                  </td>
                </tr>
                <tr
                  class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                  data-state="false"
                >
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <button
                      aria-checked="false"
                      aria-label="Select row"
                      class="peer h-4 w-4 shrink-0 rounded-sm border border-cn-primary shadow focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-cn-ring disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-cn-primary data-[state=checked]:text-cn-primary-foreground"
                      data-state="unchecked"
                      role="checkbox"
                      type="button"
                      value="on"
                    />
                  </td>
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <div
                      class="text-right font-medium whitespace-normal break-words"
                    >
                      document.pdf
                    </div>
                  </td>
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <button
                      class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-cn-primary underline-offset-4 hover:underline h-8 rounded-md px-3 text-xs"
                    >
                      <svg
                        class="lucide lucide-trash2 lucide-trash-2"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3 6h18"
                        />
                        <path
                          d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                        />
                        <path
                          d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                        />
                        <line
                          x1="10"
                          x2="10"
                          y1="11"
                          y2="17"
                        />
                        <line
                          x1="14"
                          x2="14"
                          y1="11"
                          y2="17"
                        />
                      </svg>
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="flex items-center justify-end space-x-2 py-4"
        >
          <div
            class="flex-1 text-sm text-muted-foreground"
          >
            2
             of
             
            3
             row(s) selected.
          </div>
          <div
            class="space-x-2"
          >
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-cn-input bg-cn-background text-cn-foreground shadow-sm hover:bg-cn-accent hover:text-cn-accent-foreground h-8 rounded-md px-3 text-xs"
              disabled=""
            >
              Previous
            </button>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-cn-input bg-cn-background text-cn-foreground shadow-sm hover:bg-cn-accent hover:text-cn-accent-foreground h-8 rounded-md px-3 text-xs"
              disabled=""
            >
              Next
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`singleSelectionFileTable component renders the component unchanged (snapshot) 1`] = `
<div>
  <div
    class="flex flex-col w-full h-full gap-4"
  >
    <input
      class="flex h-9 w-full rounded-md border border-cn-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
      multiple=""
      type="file"
    />
    <div
      aria-required="false"
      class="grid gap-2"
      dir="ltr"
      role="radiogroup"
      style="outline: none;"
      tabindex="0"
    >
      <div
        class="flex flex-col gap-5 w-min"
      >
        <input
          class="flex h-9 w-full rounded-md border border-cn-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm max-w-sm"
          placeholder="Search ..."
          value=""
        />
        <div
          class="rounded-md border"
        >
          <div
            class="relative w-full overflow-auto"
          >
            <table
              class="w-full caption-bottom text-sm"
            >
              <thead
                class="[&_tr]:border-b"
              >
                <tr
                  class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                >
                  <th
                    class="h-10 px-2 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  />
                  <th
                    class="h-10 px-2 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <div
                      class="text-right"
                    >
                      Filename
                    </div>
                  </th>
                  <th
                    class="h-10 px-2 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  />
                </tr>
              </thead>
              <tbody
                class="[&_tr:last-child]:border-0"
              >
                <tr
                  class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                  data-state="selected"
                >
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <button
                      aria-checked="true"
                      class="aspect-square h-4 w-4 rounded-full border border-cn-primary text-cn-primary shadow focus:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
                      data-radix-collection-item=""
                      data-state="checked"
                      role="radio"
                      tabindex="-1"
                      type="button"
                      value="0"
                    >
                      <span
                        class="flex items-center justify-center"
                        data-state="checked"
                      >
                        <svg
                          class="lucide lucide-circle h-3.5 w-3.5 fill-cn-primary"
                          fill="none"
                          height="24"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <circle
                            cx="12"
                            cy="12"
                            r="10"
                          />
                        </svg>
                      </span>
                    </button>
                  </td>
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <div
                      class="text-right font-medium whitespace-normal break-words"
                    >
                      test-file.txt
                    </div>
                  </td>
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <button
                      class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-cn-primary underline-offset-4 hover:underline h-8 rounded-md px-3 text-xs"
                    >
                      <svg
                        class="lucide lucide-trash2 lucide-trash-2"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3 6h18"
                        />
                        <path
                          d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                        />
                        <path
                          d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                        />
                        <line
                          x1="10"
                          x2="10"
                          y1="11"
                          y2="17"
                        />
                        <line
                          x1="14"
                          x2="14"
                          y1="11"
                          y2="17"
                        />
                      </svg>
                    </button>
                  </td>
                </tr>
                <tr
                  class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                  data-state="false"
                >
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <button
                      aria-checked="false"
                      class="aspect-square h-4 w-4 rounded-full border border-cn-primary text-cn-primary shadow focus:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
                      data-radix-collection-item=""
                      data-state="unchecked"
                      role="radio"
                      tabindex="-1"
                      type="button"
                      value="1"
                    />
                  </td>
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <div
                      class="text-right font-medium whitespace-normal break-words"
                    >
                      another-file.png
                    </div>
                  </td>
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <button
                      class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-cn-primary underline-offset-4 hover:underline h-8 rounded-md px-3 text-xs"
                    >
                      <svg
                        class="lucide lucide-trash2 lucide-trash-2"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3 6h18"
                        />
                        <path
                          d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                        />
                        <path
                          d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                        />
                        <line
                          x1="10"
                          x2="10"
                          y1="11"
                          y2="17"
                        />
                        <line
                          x1="14"
                          x2="14"
                          y1="11"
                          y2="17"
                        />
                      </svg>
                    </button>
                  </td>
                </tr>
                <tr
                  class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                  data-state="false"
                >
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <button
                      aria-checked="false"
                      class="aspect-square h-4 w-4 rounded-full border border-cn-primary text-cn-primary shadow focus:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
                      data-radix-collection-item=""
                      data-state="unchecked"
                      role="radio"
                      tabindex="-1"
                      type="button"
                      value="2"
                    />
                  </td>
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <div
                      class="text-right font-medium whitespace-normal break-words"
                    >
                      document.pdf
                    </div>
                  </td>
                  <td
                    class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                  >
                    <button
                      class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 text-cn-primary underline-offset-4 hover:underline h-8 rounded-md px-3 text-xs"
                    >
                      <svg
                        class="lucide lucide-trash2 lucide-trash-2"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3 6h18"
                        />
                        <path
                          d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                        />
                        <path
                          d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                        />
                        <line
                          x1="10"
                          x2="10"
                          y1="11"
                          y2="17"
                        />
                        <line
                          x1="14"
                          x2="14"
                          y1="11"
                          y2="17"
                        />
                      </svg>
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="flex items-center justify-end space-x-2 py-4"
        >
          <div
            class="flex-1 text-sm text-muted-foreground"
          >
            1
             of
             
            3
             row(s) selected.
          </div>
          <div
            class="space-x-2"
          >
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-cn-input bg-cn-background text-cn-foreground shadow-sm hover:bg-cn-accent hover:text-cn-accent-foreground h-8 rounded-md px-3 text-xs"
              disabled=""
            >
              Previous
            </button>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-cn-input bg-cn-background text-cn-foreground shadow-sm hover:bg-cn-accent hover:text-cn-accent-foreground h-8 rounded-md px-3 text-xs"
              disabled=""
            >
              Next
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
